### PR TITLE
Add config dump

### PR DESCRIPTION
**Pull request by chris-olszewski**
- [x] docs (should solve #230)
- [x] tests
- [x] redo behavior to address new specs
- [x] refactor
Addresses #226

### CODE REVIEW

**Comment by chris-olszewski** [comment]
@reconbot @southpolesteve does this look like the correct behavior?

File: src/commands/config/dump.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export function builder (yargs) {
+  return yargs
+  .pkgConf('shep', process.cwd())
+  .boolean('json')
+  .describe('json', 'Formats output as JSON')
+  .example('shep config dump beta', 'Print to console all environment variables of environment `beta` in JSON format')
+}
+
+export async function handler (opts) {
+  const envs = await load.envs()
```
Comment (southpolesteve) [inline src/commands/config/dump.js]: Is this still loading from API gateway? I think now it needs to be looking at the func aliases and deduping

Comment (chris-olszewski) [inline src/commands/config/dump.js]: I was going to switch the env behavior in another pr that I'm working on

**Comment by southpolesteve** [comment]
Cool. Can you make another issue and put it in a milestone? I wanna keep
track of stuff that needs doing before we cut 4.0

File: src/config-dump/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import * as load from '../util/load'
+import Promise from 'bluebird'
+import { getEnvironment } from '../util/aws/lambda'
+import { environmentCheck, values } from '../util/environment-check'
+
+export default async function (opts) {
+  const fns = await load.funcs()
```
Comment (chris-olszewski) [inline src/config-dump/index.js]: Should be filtering to only deployed functions

**Comment by chris-olszewski** [comment]
Going to just roll this into the larger env rework/4.0 PR

**Pull request closed without merging**